### DISCUSSION TITLE
Include Script block

### DISCUSSION THREAD

**Discussion by sameershaik1996** [postmanlabs/postman-flows#130] [Ideas]
It would be great to include a script block in which we can use javascript to add/edit variables

**Comment by samay-v** [comment]
Hey @sameershaik1996 
@saswatds Wrote a short piece on why we're choosing to not have a "Script Block" and we'd love your thoughts there. 
Check that out [here](https://github.com/postmanlabs/postman-flows/discussions/143)

**Reply by sameershaik1996** [reply]
Thanks for replying, Yes it makes more sense as to why scripts are excluded as part of this, I believe most of use these scripts in the following scenarios:

- Generate a few variables dynamically so that these can be used as part of the request, if this can be achieved using a function block that gives more control as to how we want to form the variable 
   - Write now using create Variable we can assign static values, would be great to generate dynamic variable and assign to a variable 
      which can be used across flow
- Making API call - This is can be achieved using flows 
- To trigger another postman request- Can a flow invoke another flow? Is this part of the road map

These are the ones which are on top of my mind, Let me know if these can be achieved with existing features may be I am not aware of this

**Reply by saswatds** [reply]
@sameershaik1996 Yes, we are currently working on the Modules proposal which will let you use one flow inside and another flow. After the implementation of that is complete we should be able to do everything.

**Reply by alexei-moneyr** [reply]
@samay-v it's understood what you have written, but since flows has big lack of features at least code block will help.
Modules in module will also not help a lot because of same issue - lack of features. 
Im sure one day it will have, but before this will happen, it's limited in usage. 
But still, great feature.

**Comment by horodchukanton** [comment]
+1 for a code block in any possible language.

I have a situation when the next request should use a value parsed from a header's value.
Flow have three ways to pass data: Message, Durables or ADE (modified Message) to pass variables between the requests.
None of them can be used in this case.

The only way to parse and store this value now is "Tests" and "Pre-request script" and variables.
But then, the Environments and even Collection variables are read-only in Flow context =(
So I have to use Run collection with passing the data through the collection variables.

It seems like a lot of work and confusion instead of having a small block to accept, parse and produce a Message
